### ISSUE TITLE
accessibility issue with wetboew library, empty a tags generated by wetboew js library, fails WCAG2AA

### ISSUE THREAD

**Issue by joejoseph00** [wet-boew/wet-boew#8361]
WCAG 2AA issue, empty a tags.

details in this thread:
https://www.drupal.org/project/wetkit/issues/2947343

I have isolated the issue to the wetboew js library code, the wetboew library incorrectly adds empty a tags into menu items under certain circumstances.
see screenshot:
https://www.drupal.org/files/issues/screenshot_mb-item_WCAG2AA-wet-boew.png

easy steps to reproduce: 
https://www.drupal.org/project/wetkit/issues/2947343#comment-12510758

some pretty bizarre behaviour from the wetboew library js code.

I looked through these queues here and found nothing related.

To detect these errors, use the WPSS scanning utility OR something like the docker image, the-a11ym-machine which reports WCAG2AA compliance.

**Comment by joejoseph00** [comment]
rather than implement a hack solution on our end to remove the empty a tags after the page is rendered, this should be fixed in the wetboew library it'self so that the wetboew js library doesn't create empty a tags in the first place.

**Comment by duboisp** [comment]
@joejoseph00 we would need to replicate your issue with a standalone version of WET with no drupal backend. 

The reproduction step that I need should not require me to install and configure the drupal WET variant. Also I would need to know what version of WET4 that WET drupal variant are using.

I looked at the mobile menu generated by WET from [one page](http://wet-boew.github.io/wet-boew/index-en.html) of our website and I don't see the empty anchor tag issue.

Do you have a link where I test the issue? Or can you provide the HTML code of a defective page?

**Comment by joejoseph00** [comment]
It occurs when I remove these links from our template:
 "Our Mandate"
 "Our Minister"
 "Notre Mandat"
 "Le Ministre"
 "Phone Numbers"
 "Find an employee"
 "Numéros de Téléphone"
 "Trouvez un(e) employé(e)"

The easiest way would probably be to export the page using firefox and upload a copy of it here.
When I'm back in the office I will try to get this for you.

**Comment by joejoseph00** [comment]
The source code for this is available here: 
https://buyandsell.gc.ca/standing-offers-and-supply-arrangements-application/my-agreements

if you run a wcag 2.0 test on this , the mobile will fail, the source html is fine, it is the wetboew library that adds extra empty a tags , causes fail

**Comment by joejoseph00** [comment]
go to this link
https://buyandsell.gc.ca/standing-offers-and-supply-arrangements-application/my-agreements

grab the html source code view source , the empty a tags are not there in the source
it is the wetboew library javascript that puts them there.

To view the issue, go to mobile mode click the hamburger to open the mobile menu
refer to this image to see the issue:
https://www.drupal.org/files/issues/screenshot_mb-item_WCAG2AA-wet-boew.png

**Comment by joejoseph00** [comment]
the site is currently using version v4.0.26 of the wet-boew libraries.
we're in the process of upgrading to 4.0.27 however it is not yet ready in our test environment.  After having searched through the wetboew issue queues I cannot see any mention of this issue being raised before, therefore it is very likely to still exist in 4.0.27

**Comment by EricDunsworth** [comment]
I just compared the [usability theme's content page template](http://wet-boew.github.io/themes-dist/theme-gcwu-fegc/content-en.html)'s footer against the [My agreements](https://buyandsell.gc.ca/standing-offers-and-supply-arrangements-application/my-agreements) page. The former's H3 headings are followed by lists of links, whereas the latter's H3 headings are coded as links.

My guess is that the menu plugin probably wasn't designed to accommodate linked footer headings. The plugin seems to expect those headings to be unlinked so it can turn them into ``<summary>`` elements in mobile view.

**Comment by joejoseph00** [comment]
hmm, food for thought, thanks for the input.  I still think it's pretty weird that the wetboew library jams in empty a tags.
